File: src/223/main/java/dev/nocalhost/plugin/intellij/ui/action/workload/OpenProjectExecutor.java
package dev.nocalhost.plugin.intellij.ui.action.workload;

import com.intellij.ide.RecentProjectsManagerBase;
import com.intellij.ide.impl.OpenProjectTask;

import java.nio.file.Paths;

public class OpenProjectExecutor {

    public static void open(String projectPath) {
        var task = OpenProjectTask.build();
        RecentProjectsManagerBase.getInstanceEx().openProjectSync(Paths.get(projectPath), task.withRunConfigurators());
    }
}

File: src/223/main/java/dev/nocalhost/plugin/intellij/ui/action/workload/OpenProjectExecutor.kt
package dev.nocalhost.plugin.intellij.ui.action.workload

import com.intellij.ide.RecentProjectsManagerBase
import com.intellij.ide.impl.OpenProjectTask
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import java.nio.file.Paths

class OpenProjectExecutor {
    companion object {
        @JvmStatic
        fun open(projectPath: String) {
            val task = OpenProjectTask.build()
            CoroutineScope(Dispatchers.Default).launch {
                RecentProjectsManagerBase.getInstanceEx()
                    .openProject(Paths.get(projectPath), task.withRunConfigurators())
            }
        }
    }
}
